src/puppetlabs/puppetdb/config.clj

```diff
@@ -287,56 +287,47 @@
               (format "product-name %s is illegal; either puppetdb or pe-puppetdb are allowed" product-name))))
     lower-product-name))
 
-(defn normalize-url-prefix
-  [url-prefix]
-  (cond
-   (empty? url-prefix) url-prefix
-   (.startsWith url-prefix "/") url-prefix
-   :else (str "/" url-prefix)))
-
 (defn configure-globals
   "Configures the global properties from the user defined config"
   [{:keys [global] :as config}]
   (assoc config :global
          (-> global
              (utils/assoc-when :product-name "puppetdb")
-             (update-in [:product-name] normalize-product-name)
+             (update :product-name normalize-product-name)
              (utils/assoc-when :update-server "http://updates.puppetlabs.com/check-for-updates"))))
 
-(defn warn-url-prefix-deprecation
-  [config-data]
-  (when-let [cw (get-in config-data [:global :url-prefix])]
-    (utils/println-err "The configuration item `url-prefix` in the [global] section is deprecated. It will be removed in the future."))
-  config-data)
-
-(defn warn-repl-retirement
+(defn warn-retirements
   "Warn a user they are using the old [repl] block, instead of [nrepl]."
   [config-data]
-  (when-let [cw (get-in config-data [:repl])]
+  (when (get-in config-data [:repl])
     (utils/println-err "The configuration block [repl] is now retired and will be ignored. Use [nrepl] instead. Consult the documentation for more details."))
+  (when (get-in config-data [:global :url-prefix])
+    (utils/println-err (str "The configuration item `url-prefix` in the [global] section is retired,"
+                            " please remove this item from your config."
+                            " PuppetDB has a non-configurable context route of `/pdb`."
+                            " Consult the documentation for more details."))
+    (System/exit 1))
   config-data)
 
-(defn add-web-routing-config
+(defn add-web-routing-service-config
   [config-data]
-  (let [url-prefix-path [:web-router-service :puppetlabs.puppetdb.cli.services/puppetdb-service]
-        prefix (or (get-in config-data url-prefix-path)
-                   ;; Setting url-prefix in [global] is deprecated
-                   (normalize-url-prefix (get-in config-data [:global :url-prefix]))
-                   "")
-        metrics-url-prefix-path [:web-router-service :puppetlabs.puppetdb.metrics/metrics-service]
-        metrics-prefix (get-in config-data metrics-url-prefix-path "/metrics")]
-    (-> config-data
-        (assoc-in url-prefix-path prefix)
-        (assoc-in metrics-url-prefix-path metrics-prefix))))
+  (let [default-web-router-service {:puppetlabs.puppetdb.cli.services/puppetdb-service "/pdb/query"
+                                    :puppetlabs.puppetdb.dashboard/dashboard-service "/pdb"
+                                    :puppetlabs.puppetdb.metrics/metrics-service "/metrics"}]
```

Comment:
The redirector config is in the packaging for some reasons we discussed in the standup that I'd be happy to relay! I'll throw in the newline change with any other review changes that come in